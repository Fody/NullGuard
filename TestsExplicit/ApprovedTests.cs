using Xunit.Abstractions;
using System.Linq;
using System.Threading.Tasks;
using VerifyXunit;
using Xunit;
using Fody;
using Verify;

public class ApprovedTests :
    VerifyBase
{
    [Fact]
    public Task ClassWithBadAttributes()
    {
        return Verify(Decompile("ClassWithBadAttributes"), settings);
    }

    [Fact]
    public Task ClassWithPrivateMethod()
    {
        return Verify(Decompile("ClassWithPrivateMethod"), settings);
    }

    [Fact]
    public Task ClassWithPrivateMethodNoAssert()
    {
        return Verify(Decompile("ClassWithPrivateMethod"), settings);
    }

    [Fact]
    public Task GenericClass()
    {
        return Verify(Decompile("GenericClass`1"), settings);
    }

    [Fact]
    public Task Indexers()
    {
        return Verify(Decompile("Indexers"), settings);
    }

    [Fact]
    public Task InterfaceBadAttributes()
    {
        return Verify(Decompile("InterfaceBadAttributes"), settings);
    }

    [Fact]
    public Task SimpleClass()
    {
        return Verify(Decompile("SimpleClass"), settings);
    }

    [Fact]
    public Task SimpleClassNoAssert()
    {
        return Verify(Decompile("SimpleClass"), settings);
    }

    [Fact]
    public Task SkipIXamlMetadataProvider()
    {
        return Verify(Decompile("XamlMetadataProvider"), settings);
    }

    [Fact]
    public Task SpecialClass()
    {
        return Verify(Decompile("SpecialClass"), settings);
    }

    [Fact]
    public Task PublicNestedInsideNonPublic()
    {
        return Verify(Decompile("NonPublicWithNested"), settings);
    }

    [Fact]
    public Task UnsafeClass()
    {
        return Verify(Decompile("UnsafeClass"), settings);
    }

    [Fact]
    public Task DerivedClass()
    {
        return Verify(Decompile("InternalBase.DerivedClass"), settings);
    }

    [Fact]
    public Task ImplementsInterface()
    {
        return Verify(Decompile("InternalBase.ImplementsInterface"), settings);
    }

    [Fact]
    public Task ImplementsInheritedInterface()
    {
        return Verify(Decompile("InternalBase.ImplementsInheritedInterface"), settings);
    }

    [Fact]
    public Task ImplementsInterfaceExplicit()
    {
        return Verify(Decompile("InternalBase.ImplementsInterfaceExplicit"), settings);
    }

    [Fact]
    public Task DerivedClassAssemblyBase()
    {
        return Verify(Decompile("AssemblyBase.DerivedClass"), settings);
    }

    [Fact]
    public Task ImplementsInterfaceAssemblyBase()
    {
        return Verify(Decompile("AssemblyBase.ImplementsInterface"), settings);
    }

    [Fact]
    public Task ImplementsInheritedInterfaceAssemblyBase()
    {
        return Verify(Decompile("AssemblyBase.ImplementsInheritedInterface"), settings);
    }

    [Fact]
    public Task ImplementsInterfaceExplicitAssemblyBase()
    {
        return Verify(Decompile("AssemblyBase.ImplementsInterfaceExplicit"), settings);
    }

    [Fact]
    public Task DerivedClassExternalBase()
    {
        return Verify(Decompile("ExternalBase.DerivedClass"), settings);
    }

    [Fact]
    public Task ImplementsInterfaceExternalBase()
    {
        return Verify(Decompile("ExternalBase.ImplementsInterface"), settings);
    }

    [Fact]
    public Task ImplementsInheritedInterfaceExternalBase()
    {
        return Verify(Decompile("ExternalBase.ImplementsInheritedInterface"), settings);
    }

    [Fact]
    public Task ImplementsInterfaceExplicitExternalBase()
    {
        return Verify(Decompile("ExternalBase.ImplementsInterfaceExplicit"), settings);
    }

    [Fact]
    public Task InfosList()
    {
        return Verify(AssemblyWeaver.TestResult.Messages.Select(x => x.Text), settings);
    }

    [Fact]
    public Task WarnsList()
    {
        return Verify(AssemblyWeaver.TestResult.Warnings.Select(x => x.Text), settings);
    }

    [Fact]
    public Task ErrorsList()
    {
        return Verify(AssemblyWeaver.TestResult.Errors.Select(x => x.Text), settings);
    }

    VerifySettings settings;

    public ApprovedTests(ITestOutputHelper output) :
        base(output)
    {
        settings = new VerifySettings();
        settings.UniqueForRuntime();
        settings.UniqueForAssemblyConfiguration();
        settings.AddScrubber(v => v.Replace("InternalBase.", string.Empty));
        settings.AddScrubber(v => v.Replace("AssemblyBase.", string.Empty));
        settings.AddScrubber(v => v.Replace("ExternalBase.", string.Empty));
    }

    private string Decompile(string item)
    {
        return Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, item)
            .Replace("[System.Private.CoreLib]", "[netstandard]");
    }
}