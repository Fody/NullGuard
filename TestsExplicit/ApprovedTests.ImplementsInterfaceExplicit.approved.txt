.class public auto ansi beforefieldinit ImplementsInterfaceExplicit
       extends [mscorlib]System.Object
       implements InterfaceWithAttributes
{
  .field private string '<InterfaceWithAttributes.NotNullProperty>k__BackingField'
  .method public hidebysig instance void 
          MethodWithNotNullParameter(string canBeNull,
                                     string arg) cil managed
  {
    // Code size       9 (0x9)
    .maxstack  3
    IL_0000:  ldarg.0
    IL_0001:  ldarg.1
    IL_0002:  ldarg.2
    IL_0003:  callvirt   instance void InterfaceWithAttributes::MethodWithNotNullParameter(string,
                                                                                           string)
    IL_0008:  ret
  } 
  .method public hidebysig instance string 
          MethodWithNotNullReturnValue(string arg) cil managed
  {
    // Code size       8 (0x8)
    .maxstack  2
    IL_0000:  ldarg.0
    IL_0001:  ldarg.1
    IL_0002:  callvirt   instance string InterfaceWithAttributes::MethodWithNotNullReturnValue(string)
    IL_0007:  ret
  } 
  .method public hidebysig specialname instance string 
          get_NotNullProperty() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  callvirt   instance string InterfaceWithAttributes::get_NotNullProperty()
    IL_0006:  ret
  } 
  .method public hidebysig specialname instance void 
          set_NotNullProperty(string 'value') cil managed
  {
    // Code size       8 (0x8)
    .maxstack  2
    IL_0000:  ldarg.0
    IL_0001:  ldarg.1
    IL_0002:  callvirt   instance void InterfaceWithAttributes::set_NotNullProperty(string)
    IL_0007:  ret
  } 
  .method private hidebysig newslot virtual final 
          instance void  InterfaceWithAttributes.MethodWithNotNullParameter(string canBeNull,
                                                                            string arg) cil managed
  {
    .override InterfaceWithAttributes::MethodWithNotNullParameter
    // Code size       20 (0x14)
    .maxstack  2
    IL_0000:  ldarg.2
    IL_0001:  brtrue.s   IL_0013
    IL_0003:  ldstr      "arg"
    IL_0008:  ldstr      "[NullGuard] arg is null."
    IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0012:  throw
    IL_0013:  ret
  } 
  .method private hidebysig newslot virtual final 
          instance string  InterfaceWithAttributes.MethodWithNotNullReturnValue(string arg) cil managed
  {
    .override InterfaceWithAttributes::MethodWithNotNullReturnValue
    // Code size       17 (0x11)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  dup
    IL_0002:  brtrue.s   IL_0010
    IL_0004:  pop
    IL_0005:  ldstr      "[NullGuard] Return value of method 'System.String "
    + "ImplementsInterfaceExplicit::InterfaceWithAttributes.MethodWithNotNullR"
    + "eturnValue(System.String)' is null."
    IL_000a:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_000f:  throw
    IL_0010:  ret
  } 
  .method private hidebysig newslot specialname virtual final 
          instance string  InterfaceWithAttributes.get_NotNullProperty() cil managed
  {
    .override InterfaceWithAttributes::get_NotNullProperty
    // Code size       22 (0x16)
    .maxstack  2
    IL_0000:  ldarg.0
    IL_0001:  ldfld      string ImplementsInterfaceExplicit::'<InterfaceWithAttributes.NotNullProperty>k__BackingField'
    IL_0006:  dup
    IL_0007:  brtrue.s   IL_0015
    IL_0009:  pop
    IL_000a:  ldstr      "[NullGuard] Return value of property 'System.Strin"
    + "g ImplementsInterfaceExplicit::InterfaceWithAttributes.NotNullProperty("
    + ")' is null."
    IL_000f:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_0014:  throw
    IL_0015:  ret
  } 
  .method private hidebysig newslot specialname virtual final 
          instance void  InterfaceWithAttributes.set_NotNullProperty(string 'value') cil managed
  {
    .override InterfaceWithAttributes::set_NotNullProperty
    // Code size       27 (0x1b)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  brtrue.s   IL_0013
    IL_0003:  ldstr      "value"
    IL_0008:  ldstr      "[NullGuard] Cannot set the value of property 'Syst"
    + "em.String ImplementsInterfaceExplicit::InterfaceWithAttributes.NotNullP"
    + "roperty()' to null."
    IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0012:  throw
    IL_0013:  ldarg.0
    IL_0014:  ldarg.1
    IL_0015:  stfld      string ImplementsInterfaceExplicit::'<InterfaceWithAttributes.NotNullProperty>k__BackingField'
    IL_001a:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
  .property instance string NotNullProperty()
  {
  } 
  .property instance string InterfaceWithAttributes.NotNullProperty()
  {
  } 
}
